addons/stock/models/stock_move_line.py

```diff
@@ -640,6 +641,7 @@ def _free_reservation(self, product_id, location_id, quantity, lot_id=None, pack
                     candidate.product_uom_qty = self.product_id.uom_id._compute_quantity(quantity_split, candidate.product_uom_id, rounding_method='HALF-UP')
                     move_to_recompute_state |= candidate.move_id
                     break
+            self.env['stock.move.line'].browse(to_unlink_candidates).unlink()
```

Comment:
You can even remove to_unlink_candidates.move_id from move_to_recompute_state because the unlink() of stock move line will recompute anyway their corresponding move states. 
I don't know if the impact will be really significant seeing your stats.